tools/wptrunner/wptrunner/executors/base.py

```diff
@@ -515,45 +520,63 @@ def run_test(self, test):
             for i, node in enumerate(nodes):
                 success, data = self.get_hash(node, viewport_size, dpi, page_ranges)
                 if success is False:
-                    return {"status": data[0], "message": data[1]}
+                    test_results = {"status": data[0], "message": data[1]}
+                    if self.reftest_screenshot == "always" and log_data:
+                        test_results["extra"] = {"reftest_screenshots": log_data}
+                    return test_results
 
                 hashes[i], screenshots[i] = data
                 urls[i] = node.url
 
             is_pass, page_idx = self.check_pass(hashes, screenshots, urls, relation, fuzzy)
+
             if is_pass:
+                if self.reftest_screenshot == "always":
+                    log_data.extend(
```

Comment:
So unfortunately this doesn't really work as written.

Some log formatters assume that `reftest_screenshots` is a list of exactly 3 items (e.g. https://searchfox.org/mozilla-central/source/testing/mozbase/mozlog/mozlog/formatters/tbplformatter.py#258) so we can't just add every screenshot to the log output without breaking things.

With the current API we can only really take the last screenshot we get.

Reply:
That is fine with me. I changed the function to only add the last two reftests run to the test result dictionary.